### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by rmzelle**
Once this has been reviewed and merged, we should check if the edited README.md file makes it to the "v1.0.1" branch (see also https://github.com/citation-style-language/styles/pull/5165).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,26 @@ CSL Style Repository
...
-For more information, check out [CitationStyles.org](http://citationstyles.org/) and the [repository wiki](https://github.com/citation-style-language/styles/wiki).
+For more information, check out [CitationStyles.org](https://citationstyles.org/) and the [repository wiki](https://github.com/citation-style-language/styles/wiki).
+
+Organization and distribution
+-----------------------------
+
+The CSL project typically only maintains CSL styles in the "master" branch for the latest released version of CSL.
+To provide a stable location for styles of a given CSL version, styles in "master" are released to the appropriate version branch (e.g. "v1.0.1" for as long as CSL 1.0.1 remains the latest version).
+When a new version of CSL is released (e.g. CSL 1.0.2), "master" will upgrade to that new version, and styles will now be released from "master" to a new version branch (e.g. "v1.0.2").
+Version branches for old versions of CSL will be kept but typically become completely dormant, as for these branches we generally stop maintenance, and we won't accept third-party contributions via pull requests, or backport changes made to styles in "master".
```
Comment (adam3smith) [inline README.md:18]:
```suggestion
Version branches for old versions of CSL will be kept but typically become completely dormant, as for these branches we generally stop maintenance, and we won't accept third-party contributions via pull requests, nor backport changes made to styles in "master".
```

File: README.md [hunk 2]
```diff
@@ -7,7 +7,26 @@ CSL Style Repository
...
+
+Starting with branch "v1.0.1", the version branches differ from "master" in the following ways:
+
+* the latest version branch only receives updates from "master" if all tests in "master" pass
+* `<updated/>` timestamps of styles are changed in the version branch to match the git modification date of each individual style in "master"
+* superfluous files present in "master" for e.g. style testing are removed
```
Comment (adam3smith) [inline README.md:24]:
```suggestion
* superfluous files present in "master", e.g. for style testing, are removed.
```

**Review by adam3smith — COMMENTED** [review]
Two small suggested changes, otherwise lgtm

**Comment by rmzelle** [comment]
@retorquere, looks like the GitHub Action is trying to update the timestamp of the README.md file: https://github.com/citation-style-language/styles/runs/1555065669?check_suite_focus=true

**Comment by retorquere** [comment]
This should now be fixed

**Comment by rmzelle** [comment]
(https://github.com/citation-style-language/Sheldon/commit/a7235fb01a8c7b5b8621ab264d22cf9a61ca200d)

**Comment by rmzelle** [comment]
Thanks!

**Pull request merged**